test/es-module/test-esm-resolve-type.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+// Flags: --expose-internals
+
+/**
+ * This test ensures defaultResolve returns the found module format in the
+ * return object in the form:
+ * { url: <url_value>, format: <'module'|'commonjs'|undefined> };
+ */
+
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+const fixtures = require('../common/fixtures');
+const path = require('path');
+const fs = require('fs');
+
+const assert = require('assert');
+const {
+  defaultResolve: resolve
+} = require('internal/modules/esm/resolve');
+
+/**
+ * ensure that resolving by full path does not return the format
+ * with the defaultResolver
+ */
+[ [ '/es-modules/package-type-module/index.js', undefined ],
+  [ '/es-modules/package-type-commonjs/index.js', undefined ],
+  [ '/es-modules/package-without-type/index.js', undefined ],
+  [ '/es-modules/package-without-pjson/index.js', undefined ],
+].forEach((testVariant) => {
```

Comment:
I'm a bit confused as to why these should not return the package.json "type" based format of the url in question.

Reply:
that's just because this is the current behavior of defaultResolve and this test section only ensures that this is not changed by the modifications here. Resolving by fully qualified path does not currently need to look for `package.json` and implicitly does not know the `"type"` value. The goal was not to change the current resolution behavior, just to expose the type if it was found.